python/perspective/bench/bench.py

```diff
@@ -0,0 +1,200 @@
+################################################################################
+#
+# Copyright (c) 2019, the Perspective Authors.
+#
+# This file is part of the Perspective library, distributed under the terms of
+# the Apache License 2.0.  The full license can be found in the LICENSE file.
+#
+import logging
+import os
+import sys
+import signal
+import tornado
+from timeit import timeit
+sys.path.insert(1, os.path.join(os.path.dirname(__file__), '..'))
+from perspective import Table, PerspectiveManager, PerspectiveTornadoHandler  # noqa: E402
+
+logging.basicConfig(level=logging.INFO)
+
+
+class BenchmarkTornadoHandler(tornado.web.RequestHandler):
+    """Host the results of the benchmark suite over a websocket."""
+
+    def set_default_headers(self):
+        self.set_header("Access-Control-Allow-Origin", "*")
+        self.set_header("Access-Control-Allow-Headers", "x-requested-with")
+        self.set_header('Access-Control-Allow-Methods', 'POST, GET, OPTIONS')
+
+    def get(self):
+        self.render("benchmark.html")
+
+
+class Benchmark(object):
+    """A single Benchmark function. Use as a wrapper for stateless lambdas
+    with no parameters.
+
+    Example:
+        >>> func = Benchmark(lambda: self._view.to_records(), meta={
+                "name": "to_records",
+                "group": "view"
+            })
```

Comment:
Are these for use as a decorator?  

```python
@Benchmark
def _to_records_test():
    self._view.to_records();
```

If so I think the signature of `__init__` is wrong - it wouldn't take a `func`.  I see `Benchmark` is not used as a decorator though here or in the perspective.py benchmarks themselves, but they are referenced as decorator in the docs.

Reply:
The docs are a little out of sync; `Benchmark` wraps a parameterless lambda and a metadata dictionary, which is a simpler implementation I went with instead of decorators.